crates/matrix-sdk-ui/tests/integration/timeline/mod.rs

```diff
@@ -831,8 +832,71 @@ async fn test_timeline_without_encryption_can_update() {
     assert_pending!(stream);
 }
 
+#[async_test]
+async fn test_timeline_receives_a_limited_number_of_events_when_subscribing() {
+    let room_id = room_id!("!foo:bar.baz");
+    let event_factory = EventFactory::new().room(room_id).sender(&ALICE);
+
+    let mock_server = MatrixMockServer::new().await;
+    let client = mock_server.client_builder().build().await;
+
+    mock_server.sync_joined_room(&client, room_id).await;
+
+    let event_cache = client.event_cache();
+    event_cache.subscribe().unwrap();
+
+    // The event cache contains 30 events.
+    event_cache
+        .add_initial_events(
```

Comment:
We shouldn't be using `add_initial_events` in testing, since this is going to be removed at some point. Can you use a sync instead, or if it's necessary that the event cache already contains the events, consider using an in-memory event cache store and feed it linked chunk updates that would create the events?

Reply:
This is necessary the event cache already contains the events. I'll try by feeding with linked chunk updates.